lib/parsers/dic_parser.rb

```diff
@@ -0,0 +1,38 @@
+module Parsers
+  class DicParser
+    WORD_AFF_REGEX = %r{[*\w\s]*[^\/]}
+
+    def initialize(file)
+      @file = file
+    end
+
+    def parse
+      @file.to_a
+           .map { |ln| ln.tr("\n\t", '') }
+           .yield_self do |cnt|
+             { approx_word_count: word_count_try(cnt) }.merge(fetch_words(cnt))
```

Comment:
I believe here is logic error (if the first line is not number → it is proper word, you just swallow it is unconvertible and never parse it as a word)